### PR TITLE
clientv3: integration test uses direct client calls

### PR DESCRIPTION
**Pull request by fanminshi**
clientv3 integration test was using clientv3.NewKV, clientv3.NewWatcher, etc to create specific client.
replace those with direct client calls so that the direct calls can also test grpc proxy.

### CODE REVIEW

File: clientv3/integration/watch_test.go [hunk 1]
```diff
@@ -36,9 +36,8 @@ type watcherTest func(*testing.T, *watchctx)
 
 type watchctx struct {
 	clus          *integration.ClusterV3
-	w             clientv3.Watcher
```
Comment (heyitsanthony) [inline clientv3/integration/watch_test.go]: does this and `kv` need to change? There's a bunch of renames that don't seem necessary; `Client` already implements the interface so it should be a subtype and can be directly assigned into it...

File: clientv3/integration/watch_test.go [hunk 2]
```diff
@@ -359,7 +355,7 @@ func TestWatchResumeCompacted(t *testing.T) {
 	defer clus.Terminate(t)
 
 	// create a waiting watcher at rev 1
-	w := clientv3.NewWatcher(clus.Client(0))
+	w := clus.Client(0)
 	defer w.Close()
```
Comment (heyitsanthony) [inline clientv3/integration/watch_test.go]: remove this? `clus.Terminate` should close the client

File: clientv3/integration/watch_test.go [hunk 3]
```diff
@@ -447,14 +443,14 @@ func TestWatchCompactRevision(t *testing.T) {
...
+	kv := clus.RandClient()
 	for i := 0; i < 5; i++ {
 		if _, err := kv.Put(context.TODO(), "foo", "bar"); err != nil {
 			t.Fatal(err)
 		}
 	}
 
-	w := clientv3.NewWatcher(clus.RandClient())
+	w := clus.RandClient()
 	defer w.Close()
```
Comment (heyitsanthony) [inline clientv3/integration/watch_test.go]: remove? clus.Terminate will clean this up

File: clientv3/integration/watch_test.go [hunk 4]
```diff
@@ -493,7 +489,7 @@ func testWatchWithProgressNotify(t *testing.T, watchOnPut bool) {
 	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
 	defer clus.Terminate(t)
 
-	wc := clientv3.NewWatcher(clus.RandClient())
+	wc := clus.RandClient()
 	defer wc.Close()
```
Comment (heyitsanthony) [inline clientv3/integration/watch_test.go]: remove?

File: clientv3/integration/lease_test.go [hunk 5]
```diff
@@ -466,15 +458,15 @@ func TestLeaseTimeToLive(t *testing.T) {
 	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
 	defer clus.Terminate(t)
 
-	lapi := clientv3.NewLease(clus.RandClient())
+	lapi := clus.RandClient()
 	defer lapi.Close()
```
Comment (heyitsanthony) [inline clientv3/integration/lease_test.go]: remove?

File: clientv3/integration/watch_test.go (hunk 1 above)
Comment (fanminshi) [inline clientv3/integration/watch_test.go]: yeah, that's better. should I remove the `wclient` in watchctx struct? it is not being used anywhere else.

Comment (heyitsanthony) [inline clientv3/integration/watch_test.go]: yeah, only have `w` and `kv`

File: clientv3/integration/lease_test.go [hunk 6]
```diff
@@ -261,12 +256,12 @@ func TestLeaseGrantErrConnClosed(t *testing.T) {
...
 	cli := clus.Client(0)
-	le := clientv3.NewLease(cli)
+	clus.TakeClient(0)
 
 	donec := make(chan struct{})
 	go func() {
 		defer close(donec)
-		_, err := le.Grant(context.TODO(), 5)
+		_, err := cli.Grant(context.TODO(), 5)
 		if err != nil && err != grpc.ErrClientConnClosing {
```
Comment (fanminshi) [inline clientv3/integration/lease_test.go]: the new change is failing at this part due the the err context cancelled.

```
--- FAIL: TestLeaseGrantErrConnClosed (0.19s)
	lease_test.go:266: expected grpc: the client connection is closing, got context canceled
FAIL
```

Comment (fanminshi) [inline clientv3/integration/lease_test.go]: if I close the client before the go routine, then the test passes. not sure why that happens.

Comment (heyitsanthony) [inline clientv3/integration/lease_test.go]: I think it's related to `cancelWhenStop`; the client close will trigger a different path than what was there before. Probably shouldn't `cancelWhenStop` at all, it's really old code that doesn't reflect how the client operates now.

**Comment by fanminshi** [comment]
All fixed. PTAL @heyitsanthony

**Comment by fanminshi** [comment]
Fix the clients in cluster_test.go. PTAL

**Comment by heyitsanthony** [comment]
lgtm once greenlights

**Pull request merged**